.github/renovate.json5

```diff
@@ -0,0 +1,45 @@
+{
+  $schema: "https://docs.renovatebot.com/renovate-schema.json",
+  semanticCommits: "auto",
+  enabledManagers: [
+    "helm-values",
+    "github-actions",
+    "regex"
+  ],
+  packageRules: [
+    {
+      matchDepNames: ["jenkins/inbound-agent"],
+      versioning: "loose"
+    },
+    {
+      matchManagers: [
+        "github-actions",
+        "regex"
+      ],
+      schedule: ["every monday"]
+    },
+    {
+      matchManagers: ["helm-values"],
+      bumpVersion: "minor",
+      postUpgradeTasks: {
+        "commands": ["helm unittest --strict -f 'unittests/*.yaml' charts/jenkins -u"],
```

Comment:
is this needed? won't CI just run this anyway?

Reply:
The `-u` updates the snapshots, which is needed for kiwigrid/k8s-sidecar and jenkins/inbound-agent

Reply:
ah that's cool, much simpler.

We could also configure this to update plugins too right then?
https://github.com/jenkinsci/helm-charts/blob/762105a9828e241b8c52535834e70efa2b61afa6/charts/jenkins/values.yaml#L250-L253

Reply:
Hmm, that could be a bit harder... We'd need to use the regex manager for that, so we can't use the bumpVersion functionality. I see 2 solutions for that;

- Use `postUpgradeTasks` for bumping the Chart version as well
- Use a 2-step approach, where we'd first create a draft PR using renovate and then use another GitHub Action to do stuff like bumping the Chart version, adding to the changelog, etc.

I need to verify if the postUpgradeTasks approach would work, as I don't know if things like helm (and helm unittest) are available once installed on the runner or if we'd need to build a custom Renovate image (Renovate runs in a container, but could, for example, mount some stuff).

If a custom image is required we could opt for option 2 instead. If a custom image isn't a big deal (it should be quite easy, with Renovate auto-updating it, etc.) that would also be an option.

Reply:
maybe we could maintain a `plugins.txt` file and sync it in the renovate workflow using `yq` to override what's in the values.yaml file in post upgrade tasks?

https://docs.renovatebot.com/modules/manager/jenkins/

Reply:
That would be an option yeah. That still wouldn't solve the bumpVersion issue though. PostUpgradeTasks run within the image, so we'd have to install the tools on the runner and mount the executables/binaries, use a custom image, or use the 2-step solution I proposed.

Personally, I'm more of a fan of mounting the tools since it doesn't involve custom images (maintenance!) and it doesn't require a 2-step approach.